### ISSUE TITLE
Only resolve file/remote references

### ISSUE THREAD

**Issue by whitlockjc** [apigee-127/sway#43]
When we load a Swagger document, we will fully dereference all references in the document.  For simple files, this is fine.  For files with a lot of nested objects and such, this can be a performance bottle neck.  It would be nice if Sway only resolved the remote references and then could resolve relative references on an as needed basis, preferably with some sort of cache to avoid having to reprocess references more than once.  Since `JsonRefs#resolveRefs` is always async, this would mean having to change our APIs to be async *or* we will need to update `json-refs` to have a synchronous API that only works on local references.

**Comment by whitlockjc** [comment]
This may be moot.  I'll give it a shot but to fully validate the document and to build the full object models for validation and such, I don't see how we can avoid dereferencing the full document.

**Comment by whitlockjc** [comment]
I don't think this is worth doing right now.  With `json-refs@2.0.x`, things should already be much faster and with the extra work required to make this happen, I'm not even sure there would be any benefit.  We can revisit this if anyone else has any performance related issues.

**Comment by whitlockjc** [comment]
For posterity, let me explain a few things.  This originally came up when dealing with a very poorly written Swagger document.  There was a big performance hit when validating the document.  I thought I could just resolve relative/remote references and rely on z-schema's ability to do sub-document validation and while it works, it wasn't super straight forward due to Swagger's non-body parameter objects.  Non-body parameter definitions are not a valid JSON Schema definition so Sway currently computes a JSON Schema definition for these objects.  Why is this an issue?  z-schema's sub document validation won't work with these.  This requires a workaround.  I did get this working and by itself, this likely isn't a big deal.  But there's more.

Once I got the document validation working, which really wasn't that big of a deal, I run into the problem where `ParameterValue#getSchema` and `Response#getSchema` both return fully-qualified JSON Schema values.  This means I would need to compute these, only once per object lifecycle of course, which itself adds even more complexity.  Since json-refs currently doesn't provide a `#resolveAgainst` feature, an likely never will, I would need to rewrite a good bit of this myself.  Again, by itself probably not a big deal.  But when you add this together, plus all of the internal plumbing required to remedy this, it just didn't seem like it was worth the time...especially since no one has reported performance issues.

Of course, we can revisit this if/when the time comes.
